govcd/lbserverpool_test.go

```diff
@@ -0,0 +1,105 @@
+// +build lb lbServerPool functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_LBServerPool tests CRUD methods for load balancer server pool.
+// The following things are tested:
+// Creation of load balancer server pool
+// Read load balancer server pool by both ID and Name (server pool name must be unique in single edge gateway)
+// Update - change a single field and compare that configuration and result objects are deeply equal
+// Update - try and fail to update without mandatory field
+// Delete
+func (vcd *TestVCD) Test_LBServerPool(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+	edge, err := vcd.vdc.FindEdgeGateway(vcd.config.VCD.EdgeGateway)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+
+	if !edge.HasAdvancedNetworking() {
+		check.Skip("Skipping test because the edge gateway does not have advanced networking enabled")
+	}
+
+	// Used for creating
+	lbPoolConfig := &types.LBPool{
+		Name:      Test_LBServerPool,
+		Algorithm: "round-robin",
+		Members: types.LBPoolMembers{
+			types.LBPoolMember{
+				Name:      "Server_one",
+				IpAddress: "1.1.1.1",
+				Port:      8443,
+				Weight:    1,
+				Condition: "enabled",
+			},
+			types.LBPoolMember{
+				Name:      "Server_two",
+				IpAddress: "2.2.2.2",
+				Port:      8443,
+				Weight:    2,
+				Condition: "enabled",
+			},
+		},
+	}
+
+	createdLbPool, err := edge.CreateLBServerPool(lbPoolConfig)
+	check.Assert(err, IsNil)
+	check.Assert(createdLbPool.ID, Not(IsNil))
+	check.Assert(createdLbPool.Members[0].Condition, Equals, "enabled")
```

Comment:
Should we check whether `createdLbPool.Members` is a valid array before addressing an index?

Reply:
Added.